README.md

```diff
@@ -145,7 +160,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+see main.cpp
```

Comment:
Wow, this is hard to grade.

First, you've gone further than I expected by adding a kind of lexical analysis, which is cool.

# +2

You also had some tests, but they aren't proper automated tests, because they require a human to read the output and know which should pass and fail.

# -2

Then, you decided to ignore the actual grammar (which had given words as terminals) and use your own. You version is more poetic than mine, but it meant you failed all my tests. In the end, I'm not penalizing that.

However... your state machine is a great big nested `if` statement, which is HORRIBLE! The question asked you to implement the state machine by using a table.

What's the problem with using an `if`. Well, first it isn't a table :) But second, it makes it easy for errors to creep in. Did you notice that your test 2 didn't produce any output? That's because there's no `else` for the `if` on line 34.

# -6